devito/archinfo.py

```diff
@@ -60,18 +62,36 @@ def get_cpu_brand():
             except:
                 pass
 
-            return None
+            try:
+                return cpuinfo.get_cpu_info().get('raw_arch_string')
+            except:
+                pass
 
         cpu_info['flags'] = get_cpu_flags()
         cpu_info['brand'] = get_cpu_brand()
 
-    if not all(i in cpu_info for i in ('flags', 'brand')):
-        # Fallback
-        ci = cpuinfo.get_cpu_info()
-        cpu_info['flags'] = ci.get('flags')
-        cpu_info['brand'] = ci.get('brand')
+    try:
+        if 'flags' not in cpu_info:
+            # Fallback
+            ci = cpuinfo.get_cpu_info()
+            cpu_info['flags'] = ci.get('flags')
+    except:
+        pass
 
     # Detect number of logical cores
+    try:
+        if cpu_info['brand'] == 'aarch64':
+            # In some ARM processors psutils and lscpu fail to detect cores correctly
+            logical = psutil.cpu_count(logical=True)
+            physical = psutil.cpu_count(logical=False)
+            if physical != (lscpu()['Core(s) per socket'] * lscpu()['Socket(s)']):
```

Comment:
```
if not A:
   B = A
```
=>
```
B = A
```

Reply:
True. This is a leftover from using it to print a warning message thaty was later removed.
```warning...core autodetection is not reliable....using lscpu....```